From f940fffe1ce00ba6dc1db62e12fea662a7ffc947 Mon Sep 17 00:00:00 2001
From: Mikko Ylinen <mikko.ylinen@intel.com>
Date: Wed, 24 Feb 2016 14:12:37 +0200
Subject: [PATCH] Workaround: remove handling of custom cmdline

When the BIOS is not in secure mode, the stub tries to load
an alternate command line. However, on certain boxes, this
backfires and wipes the default command line, even if an
alternate is not available.

Upstream-Status: Inapproriate

Signed-off-by: Igor Stoppa <igor.stoppa@intel.com>
Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>
---
 src/boot/efi/stub.c |   14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/boot/efi/stub.c b/src/boot/efi/stub.c
index 2cd5c33..2737fe7 100644
--- a/src/boot/efi/stub.c
+++ b/src/boot/efi/stub.c
@@ -87,20 +87,6 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
 
         cmdline_len = szs[0];
 
-        /* if we are not in secure boot mode, accept a custom command line and replace the built-in one */
-        if (!secure && loaded_image->LoadOptionsSize > 0) {
-                CHAR16 *options;
-                CHAR8 *line;
-                UINTN i;
-
-                options = (CHAR16 *)loaded_image->LoadOptions;
-                cmdline_len = (loaded_image->LoadOptionsSize / sizeof(CHAR16)) * sizeof(CHAR8);
-                line = AllocatePool(cmdline_len);
-                for (i = 0; i < cmdline_len; i++)
-                        line[i] = options[i];
-                cmdline = line;
-        }
-
         /* export the device path this image is started from */
         if (disk_get_part_uuid(loaded_image->DeviceHandle, uuid) == EFI_SUCCESS)
                 efivar_set(L"LoaderDevicePartUUID", uuid, FALSE);
-- 
1.7.10.4

